eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala

```diff
@@ -151,7 +151,7 @@ class PeerConnection(keyPair: KeyPair, conf: PeerConnection.Conf, switchboard: A
             val flags_opt = if (canUseChannelRangeQueriesEx) Some(QueryChannelRangeTlv.QueryFlags(QueryChannelRangeTlv.QueryFlags.WANT_ALL)) else None
             if (d.doSync) {
               log.info(s"sending sync channel range query with flags_opt=$flags_opt")
-              router ! SendChannelQuery(d.chainHash, d.remoteNodeId, self, flags_opt = flags_opt)
+              router ! SendChannelQuery(d.chainHash, d.remoteNodeId, self, replacePrevious = true, flags_opt = flags_opt)
```

Comment:
I wonder if we should factor this inside one single handler in state `CONNECTED` for a `MaybeSync` event?

Reply:
I don't understand what you mean and what that would bring :sweat_smile:

Reply:
This is what I have in mind, we factor the sync handling in one place.

I think we should consider dropping the support for `Features.InitialRoutingSync` alltogether, which would make this refactoring even more logical.

```diff
diff --git a/eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala b/eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala
index deedcab67..be6e88075 100644
--- a/eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala
+++ b/eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala
@@ -137,22 +137,10 @@ class PeerConnection(keyPair: KeyPair, conf: PeerConnection.Conf, switchboard: A
           d.peer ! ConnectionReady(self, d.remoteNodeId, d.pendingAuth.address, d.pendingAuth.outgoing, d.localInit, remoteInit)
           d.pendingAuth.origin_opt.foreach(_ ! ConnectionResult.Connected)

-          val canUseChannelRangeQueries = Features.canUseFeature(d.localInit.features, remoteInit.features, Features.ChannelRangeQueries)
-          val canUseChannelRangeQueriesEx = Features.canUseFeature(d.localInit.features, remoteInit.features, Features.ChannelRangeQueriesExtended)
-          if (canUseChannelRangeQueries || canUseChannelRangeQueriesEx) {
-            // if they support channel queries we don't send routing info yet, if they want it they will query us
-            // we will query them, using extended queries if supported
-            val flags_opt = if (canUseChannelRangeQueriesEx) Some(QueryChannelRangeTlv.QueryFlags(QueryChannelRangeTlv.QueryFlags.WANT_ALL)) else None
-            if (d.doSync) {
-              log.info(s"sending sync channel range query with flags_opt=$flags_opt")
-              router ! SendChannelQuery(d.chainHash, d.remoteNodeId, self, replacePrevious = true, flags_opt = flags_opt)
-            } else {
-              log.info("not syncing with this peer")
-            }
-          } else if (remoteInit.features.hasFeature(Features.InitialRoutingSync)) {
-            // "old" nodes, do as before
-            log.info("peer requested a full routing table dump")
-            router ! GetRoutingState
+          if (d.doSync) {
+            self ! DoSync
+          } else {
+            log.info("not syncing with this peer")
           }

           // we will delay all rebroadcasts with this value in order to prevent herd effects (each peer has a different delay)
@@ -356,12 +344,19 @@ class PeerConnection(keyPair: KeyPair, conf: PeerConnection.Conf, switchboard: A
         val canUseChannelRangeQueries = Features.canUseFeature(d.localInit.features, d.remoteInit.features, Features.ChannelRangeQueries)
         val canUseChannelRangeQueriesEx = Features.canUseFeature(d.localInit.features, d.remoteInit.features, Features.ChannelRangeQueriesExtended)
         if (canUseChannelRangeQueries || canUseChannelRangeQueriesEx) {
+          // This is us syncing from them
           val flags_opt = if (canUseChannelRangeQueriesEx) {
             Some(QueryChannelRangeTlv.QueryFlags(QueryChannelRangeTlv.QueryFlags.WANT_ALL))
           } else {
             None
           }
           router ! SendChannelQuery(d.chainHash, d.remoteNodeId, self, replacePrevious = false, flags_opt)
+        } else if (d.remoteInit.features.hasFeature(Features.InitialRoutingSync)) {
+          // This is them syncing from us
+          // For "old" nodes that don't support channel queries, we send them the full routing table, otherwise we will
+          // just wait for their queries
+          log.info("peer requested a full routing table dump")
+          router ! GetRoutingState
         }
         stay

```

Reply:
Done in https://github.com/ACINQ/eclair/pull/1587/commits/ea4f2a3de7e7a35e2b7d924699cfe32027323c53